File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/selection/RouteSelection.java
package rectangledbmi.com.pittsburghrealtimetracker.selection;

import java.util.Set;

import timber.log.Timber;

/**
 * An object that contains the selected route and/or all selected routes.
 *
 * Primarily used in the selection subject
 *
 * @author Jeremy Jao
 */
public class RouteSelection {

    public static RouteSelection create(Route route, Set<String> selectedRoutes) {
        return new RouteSelection(route, selectedRoutes);
    }

    private Route toggledRoute;

    private Set<String> selectedRoutes;

    private RouteSelection(Route route, Set<String> selectedRoutes) {
        toggledRoute = route;
        this.selectedRoutes = selectedRoutes;
        Timber.d("getting both routes: %s, %s", route.getRoute(), selectedRoutes.toString());
    }

    /**
     *
     * @return the toggled route
     */
    public Route getToggledRoute() {
        return toggledRoute;
    }

    /**
     *
     * @return the selected route
     */
    public Set<String> getSelectedRoutes() {
        return selectedRoutes;
    }
}

File: app/src/main/kotlin/rectangledbmi/com/pittsburghrealtimetracker/selection/RouteSelection.kt
package rectangledbmi.com.pittsburghrealtimetracker.selection

import rectangledbmi.com.pittsburghrealtimetracker.world.Route

/**
 * This is the route selection object that shows the
 * current selected route and the all routes selected
 * @author Jeremy Jao
 */
class RouteSelection(selectedRoutes: Set<String>?, toggledRoute: Route?) {
    var selectedRoutes: Set<String>? = selectedRoutes
    var toggledRoute: Route? = toggledRoute
}
